src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -107,7 +117,23 @@ private static SyntaxNode GetStartingNode(Document document, SyntaxNode root, Sn
                 startingNode = startingNode.Parent;
             }
 
-            return startingNode;
+            return true;
+        }
+
+        private static bool CaretIsInComment(SyntaxToken token, int caretPosition)
```

Comment:
💡 Consider using `SyntaxTreeExtensions.IsEntirelyWithinComment` instead of defining a new method here